src/mixins/object_geometry.mixin.js

```diff
@@ -601,77 +605,62 @@
       return cache.value;
     },
 
-    /*
+    /**
      * Calculate object dimensions from its properties
      * @private
-     * @return {Object} .x width dimension
-     * @return {Object} .y height dimension
+     * @returns {fabric.Point} dimensions
      */
     _getNonTransformedDimensions: function() {
-      var strokeWidth = this.strokeWidth,
-          w = this.width + strokeWidth,
-          h = this.height + strokeWidth;
-      return { x: w, y: h };
+      return new fabric.Point(this.width, this.height).scalarAddEquals(this.strokeWidth);
     },
 
-    /*
+    /**
      * Calculate object bounding box dimensions from its properties scale, skew.
-     * @param {Number} skewX, a value to override current skewX
-     * @param {Number} skewY, a value to override current skewY
+     * @param {Object} [options]
+     * @param {Number} [options.scaleX]
+     * @param {Number} [options.scaleY]
+     * @param {Number} [options.skewX]
+     * @param {Number} [options.skewY]
```

Comment:
adding skewX and skewY was an hack already.
It was added to try support skew indeed, and ask, how big would be the bounding box if skewing wasn't there.
And then use to counter calculate the growth of the object to follow a mouse movement.
Why are we adding scaleX and scaleY support to this?

Reply:
Because I think it is needed for fixing controls rendering for nested objects

Reply:
if we are not using those in this PR, let's not document it